### ISSUE TITLE
HookedBuildInfo doesn't support foreign-library

### ISSUE THREAD

**Issue by marknsikora** [haskell/cabal#7227]
**Describe the bug**
Trying to use a foreign-library field in a buildinfo fails the config step

**To Reproduce**

Add a buildinfo to a project like

```
foreign-library: testlib
cc-options: -I/opt/local/include/
```

```
$ cabal v2-build ...
...
Unknown field: "foreign-library"
```

**Expected behavior**
HookedBuildInfo should support all the targets in a PackageDescription

**Additional context**
Similar root issue to  #2932, reverted fix #3627

**Workaround**
Put extra flags into a file like `testlib.bi` and add a hook to manually parse this file and update the PackageDescription

**Comment by zlonast** [comment]
@sheaf Is it true that we don't plan to add this support?

```haskell
-- No support for named sublibraries, foreign libraries, tests or benchmarks,
-- because the HookedBuildInfo datatype doesn't specify what type of component
-- each component name is (so we assume they are executables).
hookedBuildInfoComponents :: HookedBuildInfo -> Set ComponentName
```

**Comment by sheaf** [comment]
Yes `HookedBuildInfo` is part of the old `UserHooks` mechanism that never got properly adapted to multiple components in the first place. It's legacy that we should delete eventually.

The recommended way to go is to use `SetupHooks`, which was designed from the start with multiple components in mind.

**Comment by zlonast** [comment]
We do not plan to add support.
